### PR TITLE
chore(ci): skip Metal E2E tests on PRs that don't touch code

### PR DESCRIPTION
**Pull request by ctreatma**

### CODE REVIEW

**Comment by ctreatma** [comment]
Made this change because I noticed the Metal E2E tests running on #766 even though that PR doesn't make any changes that would be relevant to the behavior of the Metal resources and data sources.

File: .github/workflows/metal_acctest.yml [hunk 1]
```diff
@@ -14,12 +14,10 @@ name: Metal Acceptance Tests
...
     types: [opened, synchronize, reopened, ready_for_review]
-    paths-ignore:
-      - 'LICENSE'
-      - '**.md'
-      - 'website/**'
-      - 'docs/**'
-      - '.github/ISSUE_TEMPLATE/**'
+    paths:
+      - 'go.mod'
+      - 'go.som'
```
Comment (displague) [inline .github/workflows/metal_acctest.yml:19]:
```suggestion
      - 'go.sum'
```

File: .github/workflows/metal_acctest.yml [hunk 2]
```diff
@@ -14,12 +14,10 @@ name: Metal Acceptance Tests
...
+      - '**/*.go'
```
Comment (displague) [inline .github/workflows/metal_acctest.yml:20]: Is there an option to combine this with a path-ignore of fabric/ne/...? If a PR only modified an existing NE resource, we could skip Metal tests.

Comment (ctreatma) [reply .github/workflows/metal_acctest.yml:20]: I think it's safer to assume the code might be shared

**Review by displague — APPROVED** [review]

**Pull request merged**